openg2p_registration/models/openg2p_registration.py

```diff
@@ -346,17 +369,89 @@ def create_beneficiary_from_registration(self):
             'context': context
         }
 
+    @api.onchange('stage_id')
+    def checking_duplicates(self):
+        if int(self.stage_id.id) == 2:
+            r = self.search_beneficiary()
+            if r == True:
```

Comment:
r is getting returned as status code. So, how are you determining with r == True condition? 

To check if duplicate exist, you need to iterate the response array (from search index), then save the beneficiary is from the *response*.